aptos-move/framework/aptos-framework/sources/overloadable_fungible_asset.move

```diff
@@ -0,0 +1,169 @@
+    public fun withdraw<T: key>(
+        owner: &signer,
+        store: Object<T>,
+        amount: u64,
+    ): FungibleAsset acquires OverloadFunctionStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(store));
+        let owner_address = signer::address_of(owner);
+        assert!(exists<OverloadFunctionStore>(metadata_addr), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        dispatchable_withdraw(
+            owner_address,
+            store,
+            amount,
+            &overloadable_store.transfer_ref,
+            &overloadable_store.withdraw_function,
+        )
+    }
+
+    /// Deposit `amount` of the fungible asset to `store`.
+    ///
+    /// The semantics of deposit will be governed by the function specified in OverloadFunctionStore.
+    public fun deposit<T: key>(
+        store: Object<T>,
+        fa: FungibleAsset
+    ) acquires OverloadFunctionStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(store));
+        assert!(exists<OverloadFunctionStore>(metadata_addr), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        dispatchable_deposit(
+            store,
+            fa,
+            &overloadable_store.transfer_ref,
+            &overloadable_store.deposit_function,
+        );
+    }
+
+    /// A transfer with a fixed amount debited from the sender
+    public fun transfer_fixed_send<T: key>(
+        _sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        send_amount: u64,
+    ) acquires OverloadFunctionStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(from));
+        assert!(exists<OverloadFunctionStore>(metadata_addr), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        let fa = fungible_asset::withdraw_with_ref(&overloadable_store.transfer_ref, from, send_amount);
+        deposit(to, fa);
+    }
+
+    /// A transfer with a fixed amount credited to the recipient
+    public fun transfer_fixed_receive<T: key>(
+        sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        receive_amount: u64,
+    ) acquires OverloadFunctionStore {
+        let fa = withdraw(sender, from, receive_amount);
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(from));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        assert!(fungible_asset::amount(&fa) == receive_amount, error::aborted(EAMOUNT_MISMATCH));
+        fungible_asset::deposit_with_ref(&overloadable_store.transfer_ref, to, fa);
+    }
+
+    native fun dispatchable_withdraw<T: key>(
```

Comment:
We cannot specify this function because, at the static time, `function` is unknown, and we do not know the exact behavior of this native function `dispatchable_withdraw`.

It would be useful to observe some instances of overloaded `functions` and see what properties we can specify for them.